nl_server/embeddings_map.py

```diff
@@ -37,45 +41,74 @@ class EmbeddingsMap:
   def __init__(self, embeddings_dict: dict[str, dict[str, str]]):
     self.embeddings_map: dict[str, Embeddings] = {}
 
-    indexes: List[EmbeddingsIndex] = load(embeddings_dict)
-    # Pre-load models once.
+    embeddings_info = parse(embeddings_dict)
+
+    # load models.
     self.name2model: Dict[str, EmbeddingsModel] = {}
-    for idx in indexes:
-      model_name = idx.model_name
-      if not model_name or model_name in self.name2model:
-        continue
-      elif idx.model_type == ModelType.VERTEXAI:
-        self.name2model[model_name] = VertexAIModel(model_name)
-      else:
-        self.name2model[model_name] = LocalSentenceTransformerModel(
-            idx.model_local_path)
-    for idx in indexes:
-      self._set_embeddings(idx)
+    self._load_models(embeddings_info.models)
+
+    # load indexes
+    for idx_name, idx_info in embeddings_info.indexes.items():
+      self._set_embeddings(idx_name, idx_info)
 
   # Note: The caller takes care of exceptions.
   def get(self, index_type: str = DEFAULT_INDEX_TYPE) -> Embeddings:
     return self.embeddings_map.get(index_type)
 
-  def reset_index(self, idx: EmbeddingsIndex):
-    if idx.model_name not in self.name2model:
-      self.name2model[idx.model_name] = \
-        LocalSentenceTransformerModel(idx.model_local_path)
-    self._set_embeddings(idx)
+  # Adds the new models and indexes in a embeddings_info object to the
+  # embeddings
+  def reset_index(self, embeddings_info: EmbeddingsInfo):
+    self._load_models(embeddings_info.models)
+    for idx_name, idx_info in embeddings_info.indexes.items():
+      self._set_embeddings(idx_name, idx_info)
+
+  # Loads a dict of model name -> model info
+  def _load_models(self, models: dict[str, ModelInfo]):
+    for model_name, model_info in models.items():
+      # if model has already been loaded, continue
+      if model_name in self.name2model:
+        continue
+
+      # try creating a model object from the model info
+      model = None
+      try:
+        if model_info.type == ModelType.VERTEXAI and allow_vertex_ai():
+          model = VertexAIModel(model_info)
+        elif model_info.type == ModelType.LOCAL:
+          model = LocalSentenceTransformerModel(model_info)
+      except Exception as e:
+        logging.warning(f'Skipped loading model {model_name}: {str(e)} ')
+        continue
+
```

Comment:
louder? propagate up?  and the caller can choose to ignore if custom DC, but not if main DC?

Reply:
done